src/ert/analysis/_es_update.py

```diff
@@ -702,3 +733,229 @@ def smoother_update(
+) -> None:
+    iens_active_index = np.flatnonzero(ens_mask)
+
+    progress_callback(AnalysisStatusEvent(msg="Loading observations and responses.."))
+    observations_and_responses = source_ensemble.get_observations_and_responses(
+        observations,
+        iens_active_index,
+    ).filter(pl.any_horizontal([pl.col(c).is_nan() for c in iens_active_index]))
+
+    observations_and_responses = observations_and_responses.sort(
+        by=["observation_key", "index"]
+    )
+
+    S = observations_and_responses.select(
+        observations_and_responses.columns[5:]
+    ).to_numpy()
+    observation_values = (
+        observations_and_responses.select("observations").to_numpy().reshape((-1,))
+    )
+    observation_errors = (
+        observations_and_responses.select("std").to_numpy().reshape((-1,))
+    )
+    obs_keys = (
+        observations_and_responses.select("observation_key").to_numpy().reshape((-1,))
+    )
+    indexes = observations_and_responses.select("index").to_numpy().reshape((-1,))
+
+    num_obs = len(observation_values)
+
+    smoother_snapshot.update_step_snapshots = _create_update_snapshot(
+        obs_keys=obs_keys,
+        observations=observation_values,
+        errors=observation_errors,
+        scaling=np.ones((num_obs,)),
+        ens_mean=S.mean(axis=1),
+        ens_std=S.std(ddof=0, axis=1, dtype=np.float64).astype(np.float32),
+        ens_mean_mask=np.ones(num_obs),
+        ens_std_mask=np.ones(num_obs),
+        indexes=indexes,
+    )
+
+    if num_obs == 0:
+        msg = "No active observations for update step"
+        progress_callback(AnalysisErrorEvent(error_msg=msg, data=smoother_snapshot))
+        raise ErtAnalysisError(msg)
+
+    ### EnIF ###
+    start_enif = time.time()
+
+    # Load all parameters at once
+    X_full = _all_parameters(
+        ensemble=source_ensemble,
+        iens_active_index=iens_active_index,
+    )
+    print(f"full parameter matrix shape: {X_full.shape}")
+
+    X_full_scaler = StandardScaler()
+    X_full_scaled = X_full_scaler.fit_transform(X_full.T)
+    print(f"Scaled X_full has shape: {X_full_scaled.shape}")
+
+    # Call fit: Learn sparse linear map only
+    H = linear_boost_ic_regression(
+        U=X_full_scaled,
+        Y=S.T,
+        verbose_level=5,
+    )
+
+    # Learn the precision matrix block-sparse over parameter groups
+    Prec_u = sp.sparse.csc_matrix((0, 0), dtype=float)
+    for param_group in parameters:
+        print(f"loading parameter group {param_group}")
+        config_node = source_ensemble.experiment.parameter_configuration[param_group]
+        X_local = _load_param_ensemble_array(
+            source_ensemble, param_group, iens_active_index
+        )
+        X_local_scaler = StandardScaler()
+        X_scaled = X_local_scaler.fit_transform(X_local.T)
+
+        graph_u_sub = config_node.load_parameter_graph()
+
+        # This will work for dim(X_scaled) on order O(n^5)
```

Comment:
Will likely add `fit_precision_cholesky_approximate()` to `graphite-maps`. This should scale better, also `O(n^6 or n^7)`